### PR TITLE
#23 Add exhibitions transcoder flags

### LINKED ISSUE #23: Update exhibitions transcoder settings
- [x] Add an environment variable for an `exhibitions` transcoder
- [x] Update ffmpeg settings to those below

From Trav:

> We have finally settled on settings for gallery content.
> Can we please update the Exhibitions Transcoder(s) - not Collections -  to output:
> 
> **Video**
> 
> - Wrapper: MP4
> - Codec: H264 (High Profile 4.2)
> - Bitrate: CBR (Constant Bitrate) 20Mbps
> - Resolution: 1920x1080
> - Frame Rate: 25fps
> 
> **Audio**
> 
> - Codec: AAC
> - Bitrate: 48000Hz 320kbps
> - Channel Allocation (Stereo 2Ch): Lt-Rt
> 
> Cheers,
> TRAV

Reference: https://superuser.com/a/1136305

### PR DESCRIPTION
**Pull request by sighmon**
Resolves #23

- [x] Add an environment variable for an `exhibitions` transcoder
- [x] Update ffmpeg settings to those below

Reference: https://superuser.com/a/1136305

## Testing

1. Create a `dev.env` file: `cp dev.tmpl.env dev.env`
1. Add the watch folders to your filesystem... e.g.

```
SMB_WATCH=/Users/sloffler/Code/transcoder-test-folders/watch
SMB_ACCESS=/Users/sloffler/Code/transcoder-test-folders/access
SMB_WEB=/Users/sloffler/Code/transcoder-test-folders/web
SMB_MASTER=/Users/sloffler/Code/transcoder-test-folders/master
SMB_OUTPUT=/Users/sloffler/Code/transcoder-test-folders/output
```

1. Build the project: `make build`
1. Run tests: `docker exec -it transcoder_transcoder_1 make test`
1. Stop the transcoder
1. Make a [short video file](https://acmiteam.slack.com/files/UBJ7W5H8F/F01C8RUU8BY/acmi.mp4) with the name: `B2004203_mo01_AmazingShortVideo.mp4`
1. Add it to your `watch` folder
1. Start XOS locally `docker-compose up`
1. Start the transcoder: `make up`
1. Check the video uploaded to your local XOS is the right aspect ratio, size, frames per second, and attempts to have a 20k bitrate: http://localhost:8000/admin/assets/video/?o=-5.3

### CODE REVIEW

File: app/easyaccess.py [hunk 1]
```diff
@@ -150,15 +150,54 @@ def main():
...
+            logging.info('Converting to exhibitions access format...')
+            access_metadata = convert_and_get_metadata(
+                source_file_path,
+                access_file_path,
+                settings.EXHIBITIONS_ACCESS_FFMPEG_ARGS,
+                vernon_id,
+                access_file_type,
+                title,
+            )
+            logging.info('Converting to access format... DONE\n')
```
Comment (alihaberfield) [inline app/easyaccess.py]:
```suggestion
            logging.info('Converting to exhibitions access format... DONE\n')
```

File: app/easyaccess.py [hunk 2]
```diff
@@ -150,15 +150,54 @@ def main():
...
+                    web_file_path,
+                    settings.WEB_FFMPEG_ARGS,
+                    vernon_id,
+                    web_file_type,
+                    title,
+                )
+                logging.info('Converting to web format... DONE\n')
     except Exception as e:
-        return post_slack_exception("Could not convert to access and web formats: %s" % e)
+        return post_slack_exception('Could not convert to access and web formats: %s' % e)
```
Comment (alihaberfield) [inline app/easyaccess.py:255]: What do you think about extracting our exhibitions & collector conversions to their own methods to make this a bit easier to read?

```
# CONVERT TO ACCESS AND WEB FORMATS
    try:
        if settings.EXHIBITIONS_TRANSCODER:
            # Transcoder settings for in-gallery exhibitions videos
            access_metadata, web_metadata = convert_to_exhibition_formats(access_file_path, access_file_type,
                                                                          source_file_path, title, vernon_id,
                                                                          web_file_path, web_file_type)
        else:
            # Transcoder settings for collections videos
            access_metadata, web_metadata = convert_collections_formats(access_file_path, access_file_type,
                                                                        source_file_path, title, vernon_id,
                                                                        web_file_path, web_file_type)
    except Exception as e:
        return post_slack_exception('Could not convert to access and web formats: %s' % e)

````

**Review by alihaberfield — COMMENTED** [review]
Approach looks consistent with how the transcoder is written to me. I feel like the complexity of the transcoder is outgrowing the usefulness of a single giant main() function, and opportunities could be sought to modularise it further, but I don't think that needs to be done within this ticket.

File: app/easyaccess.py (hunk 1 above)
Comment (sighmon) [reply app/easyaccess.py]: @alihaberfield totally agree. Probably good to do this as part of #20

**Review by alihaberfield — APPROVED** [review]
Looks good, thanks for the changes!

**Pull request merged**